services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/ThrottlingRerouteSpec.groovy

```diff
@@ -294,6 +294,37 @@ class ThrottlingRerouteSpec extends BaseSpecification {
         db.countFlows() == 0
     }
 
+    def "Auto-reroute on the flow which is already deleted should not revive the flow"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+        def path = PathHelper.convert(northboundService.getFlowPath(flow.id))
+
+        when: "Remove the flow"
+        northboundService.deleteFlow(flow.id)
+
+        and: "Immediately break the path to init all flows on that path to reroute"
+        def islToBreak = pathHelper.getInvolvedIsls(path).first()
+        northboundService.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+
+        then: "Flow is not present in the system after reroute timeout"
+        TimeUnit.SECONDS.sleep(rerouteDelay + WAIT_OFFSET)
+        !northboundService.getAllFlows().find { it.id == flow.id }
+        northboundService.getAllLinks().every { it.availableBandwidth == it.speed }
+
+        and: "No rule discrepancies observed"
+        topology.activeSwitches.every {
+            def rules = northboundService.validateSwitchRules(it.dpId)
+            rules.missingRules.empty && rules.excessRules.empty
+        }
+
+        and: "Bring port back up"
```

Comment:
It'd be nice to additionally ensure that port gets back up (in cleanup or something)

Reply:
In order to make sure that system was reversed to original state and next tests are allowed to run I do `Wrappers.wait(WAIT_OFFSET) { northboundService.getAllLinks().every { it.state != IslChangeType.FAILED } }` on line 325. In ensures that the port is up and the related link was again discovered